### PR TITLE
Add deepEqualAdded #18

### PR DESCRIPTION
**Pull request by jmeline**
A little more cleanup is expected but here is the first draft of this.

### CODE REVIEW

File: deep.js [hunk 1]
```diff
@@ -189,47 +189,116 @@ const deepAssignAdded = (target, ...sources) => {
...
+    );
+  }
+  return false;
 };
 
 const isObject = x => {
     return typeof x === `object` && x !== null;
 };
+
+const validateArray = (a,b) => {
```
Comment (jmeline) [inline deep.js:229]: I found that I was doing the same array looping in multiple places. I suck with naming and I am happy to take suggestions for an alternate name.

File: package.json [hunk 2]
```diff
@@ -40,16 +40,14 @@
     "env": {
       "es6": true,
       "browser": true
-    },
-    "rules": {
```
Comment (jmeline) [inline package.json:44]: I can probably leave this in.

**Review by GrosSacASac — CHANGES_REQUESTED** [review]
use 4 spaces for indentation

always include extension inside import

other than that it's ok (we can always do incremental improvements later)

File: tests/specification/deepEqual.js [hunk 3]
```diff
@@ -1,6 +1,5 @@
 import test from "ava";
-import { deepEqual, deepEqualAdded } from "../../deep.js";
-
+import { deepEqual, deepEqualAdded } from "../../deep";
```
Comment (GrosSacASac) [inline tests/specification/deepEqual.js]: include extension

**Review by GrosSacASac — CHANGES_REQUESTED** [review]

**Comment by jmeline** [comment]
Addressed concerns. If there are any others, let me know.

**Comment by GrosSacASac** [comment]
follow up issue https://github.com/GrosSacASac/utilsac/issues/31

**Pull request merged**